### DISCUSSION TITLE
Allow user comment threads on documentation website

### DISCUSSION THREAD

**Discussion by ryanscottcausecast** [godotengine/godot-proposals#4178] [General Discussions]
Documentation is no ones favorite job.  And as a result documentation consistently lags behind features.  Additionally, documentation lacks and probably always will lack robust and up to date examples or tips and tricks for implementing features.   

The fastest way to receive any sort of answers is in discord but it requires hanging around and hoping someone there can answer your question near the time you have available to hang on discord.  That's fine but its not a permanent record and doesn't work for a lot of users.  And answers are often going to be necessarily more terse as its a chat platform not a 'stack overflow'

If users could add comments below the documentation for discussions to take place we would be more easily able to make the documentation far more useful and robust without having to involve the developers.  It would also provide a resource that could be used to make the documentation above it (and in the editor) more useful without requiring a developers time.

This could super charge a new users ability to get up and running, and help everyone when they get stuck.  I think this feature would be a gigantic help to everyone.

**Comment by eon-s** [comment]
Documentation is one of the favorite jobs of the Godot's documentation team, it is curerntly up to date with features and almost up to date with API reference.

Comments on documentation are complicated because those will stay behind changes, need complex moderation, tend to be low quality and is rare to enter the page understanding everything and reach the end with enough experience to give a good answer, so it ends just filling sections with questions and no answers, that is why most modern documentations on this kind of tools just dropped the comments sections.

**Comment by Calinou** [comment]
There's already a proposal tracking this: https://github.com/godotengine/godot-proposals/issues/481

[I started an implementation a while ago](https://github.com/godotengine/godot-docs/pull/4419), but I couldn't finish it yet. I can't give an ETA for this being finished.

**Reply by eon-s** [reply]
Linking to popular Q&A results could be handy, but the Q&A also need some indication of the godot version on the results or it will be less useful than google searches.

**Reply by Calinou** [reply]
Unfortunately, the Godot version for each question is only accessible by loading the page in question, not in the list of questions. This would make scraping for those very slow, so I don't think it's viable.

**Reply by eon-s** [reply]
Yes, maybe is something to think to update before Godot 4 comes out, so questions can be tagged for 3 and before, then the new ones can be more specific.

**Comment by tetrapod00** [comment]
Implemented in https://github.com/godotengine/godot-docs/pull/7520.
